include/eve/detail/function/simd/x86/slide_right.hpp

```diff
@@ -0,0 +1,160 @@
+      }
+      else if constexpr( std::same_as<ABI,x86_256_>)
+      {
+        if constexpr( current_api >= avx2)
+        {
+          using i_t = as_integer_t<wide<T,N,ABI>>;
+          constexpr auto offset = Shift * sizeof(T);
+
+          i_t vi  = bit_cast(v, as_<i_t>{});
+          i_t bvi = _mm256_permute2x128_si256(vi, vi, 0x08);
+
+          if constexpr(offset == 16)
+          {
+            return bit_cast(bvi,as(v));
+          }
+          else  if constexpr(offset <  16)
+          {
+            return bit_cast(i_t(_mm256_alignr_epi8 (vi, bvi, 16 - offset)) , as(v));
+          }
+          else
+          {
+            return bit_cast(i_t(_mm256_bslli_epi128(bvi, offset - 16)), as(v));
+          }
+        }
+        else
+        {
+          constexpr auto shifted_bytes = sizeof(T)* Shift;
+          using f_t = typename wide<T,N,ABI>::template rebind<float>;
+          auto const w  = bit_cast(v, as_<f_t>{});
+          auto const s0 = _mm256_permute2f128_ps(w,w,0x08 );
+
+          if constexpr(shifted_bytes == 4 * 7)
+          {
+            return bit_cast(f_t(_mm256_permute_ps(_mm256_blend_ps(s0, f_t(0), 0xef), 0x39)),as(v));
+          }
+          else if constexpr(shifted_bytes == 4 * 6)
+          {
+            return bit_cast(f_t( _mm256_shuffle_ps(f_t{0}, s0, 0x40) ), as(v));
+          }
+          else  if constexpr(shifted_bytes == 4 * 5)
+          {
+            return bit_cast(f_t(_mm256_permute_ps(_mm256_blend_ps(s0, f_t(0), 0x8f), 0x93)),as(v));
+          }
+          else  if constexpr(shifted_bytes == 4 * 4)
+          {
+            return bit_cast(f_t(s0), as(v));
+          }
+          else  if constexpr(shifted_bytes == 4 * 3)
+          {
+            return bit_cast(f_t(_mm256_permute_ps(_mm256_blend_ps(s0, w, 0x11), 0x39)), as(v));
+          }
+          else  if constexpr(shifted_bytes == 4 * 2 )
+          {
+            return bit_cast(f_t(_mm256_shuffle_ps(s0, w, 0x4e)), as(v));
+          }
+          else  if constexpr(shifted_bytes == 4 )
+          {
+            return bit_cast(f_t(_mm256_permute_ps(_mm256_blend_ps(s0, w, 0x77), 0x93)), as(v));
+          }
+          else
+          {
+            if constexpr(shifted_bytes > 16 )
+            {
+              // generates [ 0 .. v0 <0 .. 0> ]
+              wide<T,N,ABI> h = _mm256_zextsi128_si256( _mm_bslli_si128 ( v.slice(lower_)
+                                                                        , shifted_bytes - 16
+                                                                        )
+                                                      );
+
+              // permutes so we have [ <0 .. 0> 0 .. v0 ]
+              auto g = bit_cast(h, as_<f_t>{});
+              return bit_cast(_mm256_permute2f128_ps(g,g,0x01), as(v));
+            }
+            else
+            {
+              // Which side are we ?
+              constexpr bool phase = Shift > (N::value/2);
```

Comment:
Can this ever be true?
We are in `avx` so total size is 32 bytes.
Therefore - `shifted_bytes > 16` covers `Shift > (N::value / 2)`